sljit_src/sljitProtExecAllocator.c

```diff
@@ -171,10 +178,12 @@ static SLJIT_INLINE int create_tempfile(void)
 	}
 
 	strcpy(tmp_name + tmp_name_len, "/XXXXXX");
+	mode = umask(0777);
```

Comment:
Is this thread safe?

Reply:
no; will better wrap within SLJIT_SINGLE_THREADED and leave the original version that used fchmod afterwards for threaded versions of the code.

note that this call is (under the sljit context) guarded by the allocator lock even in those cases.